integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ImageCatalogTests.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testCreateInvalidImageCatalogUrlNoProtocol() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "-url-np")
+                .withUrl(INVALID_IMAGECATALOG_URL), "an imagecatalog request with url no protocol"
+        );
+        when(ImageCatalog.post(), "post the imagecatalog request");
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertNotEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "-url-np");
+                }),  "check imagecatalog is not created with url no protocol");
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testCreateDeleteValidCreateAgain() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.isCreatedDeleted()
+                .withName(VALID_IMAGECATALOG_NAME + "-delete-create")
+                .withUrl(VALID_IMAGECATALOG_URL), "an imagecatalog request then delete"
+        );
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "-delete-create")
+                .withUrl(INVALID_IMAGECATALOG_URL), "same imagecatalog request again"
+        );
+        when(ImageCatalog.post(), "post the imagecatalog request");
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "-delete-create");
+                }),  "check imagecatalog is created when name was used and deleted before");
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testDeleteCbDefaultImageCatalog() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(DEFAULT_IMAGECATALOG_NAME)
+        );
+        when(ImageCatalog.delete(), "try to delete cb default image catalog");
+    }
+
+    @Test
+    public void testSetNewDeleteDefaultImageCatalog() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "default")
+                .withUrl(VALID_IMAGECATALOG_URL), "an imagecatalog request and set as default"
+
+        );
+        when(ImageCatalog.post());
+        when(ImageCatalog.setDefault());
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "default");
+                    Assert.assertEquals(imageCatalog.getResponse().isUsedAsDefault(), true);
+                    try {
+                        given(ImageCatalog.request().withName(VALID_IMAGECATALOG_NAME + "default"));
```

Comment:
Please try not to use these given/when statements in the then statement. It is hard to understand, and also mess up the test context: a next then statement on the image catalog will get the new imagecatalog request - not the original one what you probably want to assert... hard to follow.